### ISSUE TITLE
[Bug]: vscode autoimport not working with `pnpIgnorePatterns`

### ISSUE THREAD

**Issue by andreialecu** [yarnpkg/berry#3816]
### Self-service

- [X] I'd be willing to implement a fix

### Describe the bug

This is a recent regression I think, because it used to work.

I have a hybrid pnp/node_modules project root, where I'm using angular with the nm linker. The angular project is under a `nm-packages` directory and it is added to `pnpIgnorePatterns`.

Auto import is not working, and the tsserver.log has the reason for it:

```
Exception on executing command {"seq":26,"type":"request","command":"getCodeFixes","arguments":{"file":".../src/app/shared/services/players.service.ts","startLine":73,"startOffset":26,"endLine":73,"endOffset":37,"errorCodes":[2304]}}:

    Cannot read property 'findPackageLocator' of null

    TypeError: Cannot read property 'findPackageLocator' of null
        at tryGetModuleNameAsNodeModule (/.yarn/cache/typescript-patch-e9a14bf896-2f915f8ac1.zip/node_modules/typescript/lib/tsserver.js:114820:38)
        at Object.getModuleSpecifiers (/.yarn/cache/typescript-patch-e9a14bf896-2f915f8ac1.zip/node_modules/typescript/lib/tsserver.js:114491:33)
        at /.yarn/cache/typescript-patch-e9a14bf896-2f915f8ac1.zip/node_modules/typescript/lib/tsserver.js:142464:44
        at Object.flatMap (/.yarn/cache/typescript-patch-e9a14bf896-2f915f8ac1.zip/node_modules/typescript/lib/tsserver.js:515:25)
        at getNewImportFixes (/.yarn/cache/typescript-patch-e9a14bf896-2f915f8ac1.zip/node_modules/typescript/lib/tsserver.js:142463:23)
        at getFixesForAddImport (/.yarn/cache/typescript-patch-e9a14bf896-2f915f8ac1.zip/node_modules/typescript/lib/tsserver.js:142475:66)
        at getImportFixes (/.yarn/cache/typescript-patch-e9a14bf896-2f915f8ac1.zip/node_modules/typescript/lib/tsserver.js:142337:63)
```

### To reproduce

Repro at:

https://github.com/andreialecu/repro-yarn-nm-pnp

### Environment

```shell
➜ yarn dlx -q envinfo --preset jest

  System:
    OS: macOS 12.0.1
    CPU: (8) arm64 Apple M1
  Binaries:
    Node: 16.13.0 - /private/var/folders/9p/k1yqxx0d7rn1nlztg_wm7sbw0000gn/T/xfs-43ab1ee2/node
    Yarn: 3.1.0 - /private/var/folders/9p/k1yqxx0d7rn1nlztg_wm7sbw0000gn/T/xfs-43ab1ee2/yarn
    npm: 8.1.0 - ~/.nvm/versions/node/v16.13.0/bin/npm
```
```


### Additional context

_No response_

**Comment by alamothe** [comment]
Hi, is this published yet? 

I'm having a similar issue, but can't find any error in the logs, so not sure if it is the same. Thanks!

**Comment by merceyz** [comment]
It's published as `3.2.0-rc.7`
```sh
yarn set version canary
```

**Comment by alamothe** [comment]
Auto-import still not working with `3.2.0-rc.7`. Should I file a new issue?
